app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -162,10 +167,23 @@ def os_disk_caching
       @azure_vm.storage_profile.os_disk.caching
     end
 
-    def script_command
+    def script_uris
+      return @script_uris unless @azure_vm.resources
+      ext_name = @azure_vm.resources.first.id.split('/')[-1]
+      ext = sdk.get_vm_extension(@azure_vm.resource_group, name, ext_name)
```

Comment:
Can you memoize the result? If you already have the extension, and you want to call `script_command` too - there is no need to get the extension yet again.

Reply:
done